cli/src/oxide_override.rs

```diff
@@ -0,0 +1,600 @@
+            };
+            let Some(name) = item_name.strip_prefix("#/definitions/") else {
+                unimplemented!("returned array type was not a reference");
+            };
+            name.to_string()
+        }
+        SingleOrVec::Vec(_) => {
+            unimplemented!("nested array of items")
+        }
+    };
+
+    let ref_fields = get_referenced_type_fields(root_schema, &item_type);
+    (ref_fields, ReturnType::Array)
+}
+
+/// Find the fields present on individual items in the array type of a `list_start`.
+fn list_start_fields(root_schema: &RootSchema) -> Vec<String> {
+    let props = &root_schema.schema.object.as_ref().unwrap().properties;
+
+    let items = props
+        .get("items")
+        .expect("list response type does not have items")
+        .clone()
+        .into_object()
+        .array
+        .unwrap()
+        .items
+        .clone()
+        .unwrap();
+
+    let item_type = match items {
+        schemars::schema::SingleOrVec::Single(s) => {
+            let item_name = s.into_object().reference.clone().unwrap();
+            item_name
+                .strip_prefix("#/definitions/")
+                .unwrap()
+                .to_string()
+        }
+        schemars::schema::SingleOrVec::Vec(_) => {
+            unimplemented!("nested array of items")
+        }
+    };
+
+    get_referenced_type_fields(root_schema, &item_type)
+}
+
+/// Get the field names of a type using its name.
+fn get_referenced_type_fields(root_schema: &RootSchema, type_name: &str) -> Vec<String> {
+    let item_schema = root_schema
+        .definitions
+        .get(type_name)
+        .unwrap()
+        .clone()
+        .into_object();
+
+    // Generally child items are a single type, but a few APIs return an enum, e.g.
+    // `/v1/anti-affinity-groups/{anti_affinity_group}/members`. In this case we collect all of the
+    // fields present on the variants.
+    match (
+        item_schema.object.as_ref(),
+        item_schema.subschemas.and_then(|s| s.one_of),
+    ) {
+        (Some(obj), _) => obj.properties.keys().cloned().collect(),
+        (_, Some(variants)) => collect_variant_fields(&variants),
+        _ => unimplemented!("item was neither an object nor one_of subschema"),
+    }
+}
+
+/// Gather the fields present on each variant, removing duplicates and retaining their original order.
+fn collect_variant_fields(variants: &[Schema]) -> Vec<String> {
+    let mut fields = IndexSet::new();
+    for variant in variants {
+        fields.extend(
+            variant
+                .clone()
+                .into_object()
+                .object
+                .as_ref()
+                .expect("variant was not an object")
+                .properties
+                .keys()
+                .cloned(),
+        );
+    }
+    fields.into_iter().collect()
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_table_schema_parsing() {
```

Comment:
My fear is that this test is going to break frequently as a result of unexpected schema data. Is that a rational fear or do you expect that this will be fairly robust?